activemodel/lib/active_model/secure_password.rb

```diff
@@ -97,7 +121,20 @@ def has_secure_password(attribute = :password, validations: true)
         #   user.authenticate_password('mUc3m00RsqyRe') # => user
         define_method("authenticate_#{attribute}") do |unencrypted_password|
           attribute_digest = send("#{attribute}_digest")
-          BCrypt::Password.new(attribute_digest).is_password?(unencrypted_password) && self
+          password = BCrypt::Password.new(attribute_digest)
+          if password.is_password?(unencrypted_password)
+            if password.cost < ActiveModel::SecurePassword.cost
+              rehashed_password = BCrypt::Password.create(unencrypted_password, cost: ActiveModel::SecurePassword.cost)
```

Comment:
Here's the auto-pw-rehash bit. If the pw matches and has a lower cost, rehash and save it.